### PR TITLE
TRT-2787: Upgrade ESLint 8→9, Prettier 2→3, Vitest 3→4 to fix npm deprecation warnings

### PR DESCRIPTION
**Pull request by smg247**
All four npm deprecation warnings from `make npm` are resolved:
- @humanwhocodes/config-array and @humanwhocodes/object-schema (ESLint 8 deps)
- eslint@8.57.1 (end-of-life)
- glob@10.5.0 (vitest 3 transitive dep)

Migrates to ESLint 9 flat config (eslint.config.mjs), adds js.configs.recommended with no-unused-vars set to warn, and fixes several bugs caught by the new rules:
- !(date instanceof Date) operator precedence fix in helpers.jsx
- filter.items === [] always-false comparison in helpers.jsx and JobAnalysis.jsx
- _.filter → lodash.filter undefined variable in IntervalsChart.jsx
- hasOwnProperty → Object.hasOwn in CompReadyVars.jsx and TriageFields.jsx

Also removes unused devDependencies (eslint-plugin-node, eslint-plugin-import, eslint-plugin-promise, prettier-eslint) and renames config files to .mjs to eliminate the Vite CJS deprecation warning.

## Summary by CodeRabbit

* **Bug Fixes**
  * Job/test/API links now omit empty filter parameters so URLs stay clean and consistent.
  * Improved relative time handling to avoid mis-detecting non-Date inputs.
  * Interval chart filtering now reliably matches the correct results.
* **Chores**
  * Updated linting and formatting setup (ESLint flat config, Prettier ESM) and refreshed Vite/Vitest configuration.
  * Upgraded related tooling and adjusted minor code formatting/cleanup across UI components.

### CODE REVIEW

File: sippy-ng/package.json [hunk 1]
```diff
@@ -77,21 +73,19 @@
...
-    "eslint-plugin-prettier": "^3.4.1",
-    "eslint-plugin-promise": "^6.6.0",
+    "eslint": "^9.0.0",
+    "eslint-config-prettier": "^10.0.0",
+    "eslint-plugin-prettier": "^5.0.0",
     "eslint-plugin-react": "^7.24.0",
     "eslint-plugin-sort-imports-es6-autofix": "^0.6.0",
-    "eslint-plugin-unused-imports": "^3.2.0",
+    "eslint-plugin-unused-imports": "^4.0.0",
+    "globals": "^16.0.0",
```
[the remark this answers is not in the document]
Comment (smg247) [reply sippy-ng/package.json:85]: We don't pin here. we just use the lock file.

File: sippy-ng/src/component_readiness/CompReadyTestPanel.jsx [hunk 2]
```diff
@@ -236,18 +236,20 @@ export default function CompReadyTestPanel(props) {
...
+              {printParamsAndStats(
+                'Sample (being evaluated)',
+                data.sample_stats,
+                data.sample_stats.Start.toString(),
+                data.sample_stats.End.toString(),
+                loadedParams.variantCrossCompare,
+                loadedParams.compareVariantsCheckedItems
+              )}
+            </Grid>
+          )}
```
[the remark this answers is not in the document]
Comment (smg247) [reply sippy-ng/src/component_readiness/CompReadyTestPanel.jsx:252]: We don't have much testing on the frontend, that is another issue entirely. not adding it in this pr

File: sippy-ng/src/helpers.jsx [hunk 3]
```diff
@@ -70,7 +70,7 @@ export function formatDateToSeconds(dateString) {
 // This is because the ES6 Intl.RelativeTime isn't available in all environments yet,
 // e.g. Safari and NodeJS.
 export function relativeTime(date, startDate) {
-  if (!date instanceof Date) {
+  if (!(date instanceof Date)) {
     date = new Date(date)
```
[the remark this answers is not in the document]
Comment (smg247) [reply sippy-ng/src/helpers.jsx:74]: We don't have much testing on the frontend, that is another issue entirely. not adding it in this pr

**Comment by smg247** [comment]
/override-sticky ci/prow/unit

It won't pass until the buildroot changes in this PR go in

File: Dockerfile [hunk 4]
```diff
@@ -3,7 +3,7 @@ WORKDIR /go/src/sippy
 COPY . .
 ENV PATH="/go/bin:${PATH}"
 ENV GOPATH="/go"
-RUN dnf module enable nodejs:18 -y && dnf install -y go make npm && make build
+RUN dnf module enable nodejs:20 -y && dnf install -y go make npm && make build
```
[the remark this answers is not in the document]
Comment (smg247) [reply Dockerfile:6]: This is a multi-stage build — the builder stage is only used to compile binaries, and only specific files are COPY --from=builder into the final base image. The DNF cache never makes it into the shipped image layer, so cleaning it adds build time for no size benefit. The Dockerfile.buildroot already does this cleanup because it is the final image used in CI.

**Comment by not-stbenjam** [comment]
## Deep Review Verdict (Grok 4.5)

**Disposition**: REQUEST_CHANGES — confirmed CI break: Vitest 4 cannot run on the Node 18 unit image because the `Dockerfile.buildroot` Node 20 bump never reaches CI.

### Panel Synthesis

Bugs and security are clean. Multiple specialists (adversarial, architecture, consistency, QA) independently flagged the empty-filter null-safety gap and the stale Prettier README path. The merge blocker is the reproduced CI failure: Vitest 4 requires Node 20+, but OpenShift CI `sparse_checkout_files` for the unit root image omits `Dockerfile.buildroot`, so the PR's `nodejs:20` change never builds and unit stays on Node 18 (`node:inspector/promises` missing). Until that is fixed (release-config sparse checkout, alternate Node install in the test path, or stay on Vitest 3.x), `ci/prow/unit` cannot pass without override.

### Required Actions Before Merge

1. **Unblock CI Node for Vitest 4** (`Dockerfile.buildroot` / openshift/release sparse checkout, or temporary Vitest 3 pin / in-job Node ≥20 install). Confirmed via reproducer against unit job [2080272974325223424](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_sippy/3812/pull-ci-openshift-sippy-main-unit/2080272974325223424).
2. **Harden empty-filter guards** in `sippy-ng/src/helpers.jsx` (and `JobAnalysis.jsx`) to `!filter || !filter.items || filter.items.length === 0` (or `!filter?.items?.length`).
3. **Fix docs drift**: `sippy-ng/README.md` link `prettier.config.js` → `prettier.config.mjs`.

<details>
<summary>Reproducer: Vitest 4 / CI Node 18 / sparse_checkout</summary>

**Steps:** Verified vitest@4.1.10 engines (`^20 || ^22 || >=24`); Node 18 fails `import('node:inspector/promises')`; unit job sparse_checkout is only `.devcontainer/Dockerfile`, `Dockerfile`, `chat/Dockerfile` (no `Dockerfile.buildroot`); sparse-sim of PR overlay leaves buildroot on `nodejs:18` while `Dockerfile` shows `nodejs:20`; prow root + unit logs show Node 18.20.8 and the inspector crash.

**Expected:** Unit root image builds from PR `Dockerfile.buildroot` with Node 20 so Vitest 4 runs.

**Actual:** Root rebuild uses base main buildroot (Node 18); `make test` crashes with `Error: No such built-in module: node:inspector/promises`.

</details>

### Optional Follow-ups

- Add `engines.node: ">=20"` and a short Node 20 note in `sippy-ng/README.md`.
- Add Vitest regression tests for `relativeTime` and empty-filter path helpers.
- Declare `lodash` as a direct dependency or replace with native `Array` methods.
- Merge duplicate Vitest ESLint config blocks; consider `ecmaVersion: 2022`.
- Guard invalid dates in `relativeTime` after coercion (`Number.isNaN(date.getTime())`).

---

<details>
<summary>Specialist Findings</summary>

**Bugs**: No functional regressions found in current call sites. The claimed fixes (`!(date instanceof Date)`, empty-filter URL paths, `lodash.filter`, `Object.hasOwn`) are real bugfixes.

**Adversarial**:
1. **BLOCKING (reproduced)**: Vitest 4 dies on CI Node 18; `Dockerfile.buildroot` Node 20 bump never applies.
2. **SUGGESTION**: `filter.items.length === 0` throws when `filter` is truthy but `items` is missing (`{}`).
3. **SUGGESTION**: `relativeTime` now coerces `null` / invalid strings into misleading relative times instead of throwing.
4. **NOTE**: `lodash` remains an undeclared direct dependency in `IntervalsChart.jsx`.

**Security & Supply Chain**: No issues. Checked Node bump, `npm ci --ignore-scripts`, lockfile, ESLint plugin removals, and Vite/Vitest/Prettier upgrades (dev-only).

**Architecture**:
- Align shared path helpers on a null-safe empty-filter check (`!filter?.items?.length`).
- Collapse duplicate Vitest blocks in `eslint.config.mjs`.
- README still points at removed `prettier.config.js`.
- Consider encoding Node 20+ via `package.json` `engines`.

**Codebase Consistency**: Same README / Node engines / lodash / ESLint Vitest-block notes; bugfix patterns otherwise match existing conventions.

**QA Engineer**: No regression tests for the four claimed logic fixes. Highest value: `helpers.test.jsx` covering `relativeTime` coercion and empty-filter URL builders; extract/test `filterIntervals` / `withoutJobRunFilters` if practical.

**Technical Writer**: Update `sippy-ng/README.md` Prettier config link to `prettier.config.mjs`. Optionally document Node 20+ for the frontend toolchain.

</details>

<details>
<summary>Stats</summary>

Arbiter summary: ~15 findings from 7 specialists.
Kept: 1 blocking (1 reproduced), ~10 suggestions/notes.
Dropped: several duplicates (empty-filter, README prettier path, engines/Node, lodash) merged across specialists.
False positives: none material; bugs specialist correctly noted current call sites supply `items`, but adversarial/QA still correctly flag the weaker contract.

</details>

<sub>Generated by /deep-review</sub>

**Comment by stbenjam** [comment]
Above is Grok, ignore him - it doesn't realize the buildroot changes won't be tested in CI.

/lgtm
/override ci/prow/unit
